Sming/Components/crypto/include/Crypto/HashContext.h

```diff
@@ -31,17 +31,17 @@ template <class Engine_> class HashContext
 		uint64_t count;
 	};
 
-	HashContext()
+	template <typename... EngineArgs> HashContext(EngineArgs&&... engineArgs)
 	{
-		reset();
+		reset(std::forward<EngineArgs>(engineArgs)...);
 	}
 
 	/**
 	 * @brief Reset the context for a new calculation
 	 */
-	HashContext& reset()
+	template <typename... EngineArgs> HashContext& reset(EngineArgs&&... engineArgs)
 	{
-		engine.init();
+		engine.init(std::forward<EngineArgs>(engineArgs)...);
 		return *this;
 	}
```

Comment:
I'll try to make the hash size a template parameter (probably by adding a templated layer on top of the current engine to avoid an all-header implementation).
However, the argument forwarding is also needed for the "keyed" mode, see the `Blake2sEngine::init(const Secret& key, size_t hashSize)` overload.

Reply:
I've pushed some suggestions to your repo. Working well my end now :-)

Reply:
Thanks for your help!
I did some minor tweaks on top of that.

This Codacy tool is driving me nuts. The warning is a false positive and the blue stuff is by design. Is there any justification syntax I can use?